plugin/src/test/java/io/jenkins/plugins/casc/vault/VaultSecretSourceTest.java

```diff
@@ -143,6 +149,31 @@ public void kv2WithApproleMultipleKeysOverriden() {
         assertThat(SecretSourceResolver.resolve(context, "${key1}"), equalTo("123"));
     }
 
+    @Test
+    public void kv2WithApproleWithReauth() {
+        envVars.set("CASC_VAULT_APPROLE", VAULT_APPROLE_ID);
+        envVars.set("CASC_VAULT_APPROLE_SECRET", VAULT_APPROLE_SECRET);
+        envVars.set("CASC_VAULT_PATHS", VAULT_PATH_KV2_AUTH_TEST);
+        envVars.set("CASC_VAULT_ENGINE_VERSION", "2");
+        assertThat(SecretSourceResolver.resolve(context, "${key1}"), equalTo("auth-test"));
+
+        try {
+            // Wait for auth token to become stale
+            Thread.sleep(5000);
```

Comment:
maybe this is a bit long? can you make the token valid for a lot less?

Reply:
Reduced to 2s